### PR TITLE
Fix TextChoice options sometimes creating a broken YAML

### PR DESCRIPTION
**Pull request by LegendaryLinux**
## What is this fixing or adding?
Fixes a bug reported here: https://github.com/ArchipelagoMW/Archipelago/issues/3367

TextChoice options' custom values were not being properly accounted for in player-options. They are now being properly filled and the offending key is removed.

## How was this tested?
Ran WebHost locally and generated several Tunic YAML files.

## If this makes graphical changes, please attach screenshots.
It does not.

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]
Fixed the issue

File: WebHostLib/options.py [hunk 1]
```diff
@@ -173,16 +173,23 @@ def generate_yaml(game: str):
...
+            # Detect and build ItemDict options from their name pattern
             if key_parts[-1] == "qty":
                 if key_parts[0] not in options:
                     options[key_parts[0]] = {}
                 if val != "0":
                     options[key_parts[0]][key_parts[1]] = int(val)
                 del options[key]
 
+            # Detect keys which end with -custom, indicating a TextChoice with a possible custom value
+            if key_parts[-1].endswith("-custom"):
```
Comment (Berserker66) [inline WebHostLib/options.py:187]:
```suggestion
            elif key_parts[-1].endswith("-custom"):
```

Comment (LegendaryLinux) [reply WebHostLib/options.py:187]: I don't believe there was any possible form submission which might trigger both, but your suggestion is logically correct.

**Review by NewSoupVi — APPROVED** [review]
Looks correct

**Pull request merged**